### ISSUE TITLE
AttributeError: Can't pickle local object 'schedule.<locals>.schedule_fn'

### ISSUE THREAD

**Issue by tchaton** [pytorch/pytorch#64492]
## To Reproduce

When using PyTorch Kineto Profiler with PyTorch Lightning and ddp_spawn accelerator, 
the schedule_fn is not pickable https://github.com/pytorch/pytorch/blob/master/torch/profiler/profiler.py#L24.

Solution, take it out of the function.

1.
1.
1.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by orionr** [comment]
cc @robieta @chaekit as well as the above. Also just added both of you to the oncall at https://github.com/pytorch/pytorch/issues/24422. Is this something we should tag distributed folks on?

**Comment by robieta** [comment]
Yeah, the issue is that pickle works by first resolving the import path of the object, and then uses that as a schema to load the bytes. So in this case it can't get at the appropriate context (because it's <locals>) and the unpickle fails. I'll throw up a PR to fix it by turning `schedule` into a functor.

**Comment by MaximovaIrina** [comment]
Chao, everyone!
I had the same problem.
I see you opened MR with the fixes, but you haven't merged it.


When will these changes be available in the master?

**Comment by MaximovaIrina** [comment]
P.S. I copied these changes into my local torch package and everything worked. Wish they had gone to the master.
